tools/jit/gen_unboxing.py

```diff
@@ -0,0 +1,182 @@
+            # Note [The ATen Codegen Unboxing API]
+            # Similar to the ATen Operators API, ATen Codegen Unboxing API lives in the at::unboxing namespace, and
+            # will be used by codegen unboxing wrappers (CodegenUnboxingWrappers.cpp).
+            # The Wrappers will be registered into torch::jit::OperatorRegistry using RegisterOperators API.
+            #
+            # Important characteristics about the Codegen Unboxing API:
+            # (1) It follows the OperatorRegistry API.
+            #     This is kind of necessary to avoid overhead.
+            #     For example: if it followed the C++ API, then all of the faithful C++ factory functions
+            #     would need to wrap their arguments into TensorOptions only to unwrap them again.
+            # (2) Under the hood it calls C++ API.
+            return f"""
+// aten::{f.func}
+TORCH_API void {f.func.name.unambiguous_name()}(Stack & stack);
+"""
+        else:
+            sig_group = CppSignatureGroup.from_native_function(
+                f, method=(Variant.method in f.variants)
+            )
+            sig = sig_group.most_faithful_signature()
+            # parse arguments into C++ code
+            binding_list, code_list = convert_arguments(f)
+
+            # for each C++ argument, generate the conversion code
+            code_connector = "\n\t"
+            arg_connector = ", "
+            # function call and push back to stack
+            prefix = "self_base." if sig.method else "at::"
+            translated_args = translate(binding_list, sig.arguments(), method=sig.method)
+            args_str = f"{arg_connector.join(e.expr for e in translated_args)}"
+            if len(f.func.returns) == 0:
+                ret_str = ""
+                push_str = ""
+            else:
+                ret_str = "auto result_ = "
+                push_str = """
+    pack(stack, std::move(result_));
+                """
+            return f"""
+// aten::{f.func}
+TORCH_API void {f.func.name.unambiguous_name()}(Stack & stack) {{
+    {code_connector.join(code_list)}
+
+    drop(stack, {len(binding_list)});
+
+    {ret_str}{prefix}{sig.name()}({args_str});
+    {push_str}
+}}
+"""
+
+
+# Generates RegisterCodegenUnboxedKernels.cpp.
+@dataclass(frozen=True)
+class ComputeCodegenUnboxedKernels:
+    @method_with_native_function
+    def __call__(self, f: NativeFunction) -> str:
+        # We unconditionally generate function wrappers,
+        sig_group = CppSignatureGroup.from_native_function(
+            f, method=(Variant.method in f.variants)
+        )
+
+        sig = sig_group.most_faithful_signature()
+
+        # escape double quote in schema, get rid of extra double quotes
+        schema = json.dumps(sig.func.__str__())[1:-1]
+
+        return f"""
+OperatorGenerator(
+    TORCH_SELECTIVE_SCHEMA("aten::{schema}"),
+    [](Stack & stack) {{
+        RECORD_FUNCTION("{sig.name()}", std::vector<c10::IValue>());
```

Comment:
Hmm, how come there needs to be a RECORD_FUNCTION here? I don't see it in the "traditional" codepath

Reply:
For mobile side profiling purpose. I can add them in a different PR if you think it's cleaner that way

Reply:
No I don't need it to be in a separate PR, I just need to know what in this path is parity with server side, and what is different. For example, having an xref saying "this needs to get synchronized with that" would go a long way for making sure divergence doesn't happen.